### DISCUSSION TITLE
CircleCI - Private Orbs

### DISCUSSION THREAD

**Discussion by LukeMathWalker** [renovatebot/renovate#11671] [Request Help] [answered]
I have been playing around with the `circleci` manager - it looks great, but I can't figure out how to make it work with private orbs: CircleCI returns a 404, but there are no instructions on how to set credentials to see private orbs.

Are they supported?

**Comment by rarkins** [comment]
I didn't even know there were private orbs! If it's the same URL and uses something simple like a Bearer token then you can perhaps get it to work with a `hostRules` entry, but otherwise if it's a custom host or more complicated authentication then it would need a feature request and we document how it would work

**Comment by LukeMathWalker** [comment]
I am trying to find the documentation for the GraphQL API you seem to be calling for CircleCI, but with no luck 😅

**Comment by LukeMathWalker** [comment, marked as answer]
Ok, found how to do it - it requires an `Authorization` header set to the CircleCI API token (no `Bearer` prefix). 
Is there a way to make that happen via configuration in `hostRules` or does it require a code change?

**Reply by viceice** [reply]
https://docs.renovatebot.com/configuration-options/#authtype

**Reply by rarkins** [reply]
```
{
  "hostRules": [
    {
      "matchHost": "https://circleci.com",
      "token": "<CircleCI token>",
      "authType": "Token-Only"
    }
  ]
}
```

**Reply by LukeMathWalker** [reply]
That solved it, thanks a lot!
